### ISSUE TITLE
5.5.6 - MoneroV7 - MinGW - Windows

### ISSUE THREAD

**Issue by LightningW9** [genesismining/sgminer-gm#125]
Thanks for keeping this miner updated. I appreciate it.

Monerov7 is working fine with mingw build. Visual studio build looks very outdated.

With the v7 switch there is a bit of spam with "switch to monero variant" at startup for some reason. I have added "monero":true to my config file.

There were some minor issues with the build in mingw

- rand_s error in sgminer.c - code commented out - used for ethnonce - may break ETHASH (I don't mine it anyway)
- mul128 asm error in cryptonight.c - instead using inline function from https://github.com/bearbin/cpuminer-cryptonight/blob/master/cryptonight_lobotomized.c
- @LIBCURL_CFLAGS@ error from makefile.am - removed to compile

Dependencies:
pthreads-w32-2.9.0-mingw32-pre-20110507-2-dev
libpdcurses-3.4-1-mingw32-dev
PDCurses-3.4-1-mingw32-bin
amd app sdk 2.9
amd adl sdk 6.0
gtk2-runtime-2.24.10-2012-10-10-ash
pkg-config_0.26-1_win32
pkg-config-dev_0.26-1_win32
curl-7.39.0-devel-mingw32
jansson-2.7

If anyone wants to test my build, you can use it at your own risk here: https://mega.nz/#!vhtgGIST!zxN8epOvSqvmlyJv5b0dahfL9PdyW-eLXHi_5YGpQ9Q

**Comment by ku4eto** [comment]
Thanks, building this seems more PITA than the Cygwin one, i didnt bothered at all. Even if i did build Cygwin executable, it was shutting down almost instantly, something went wrong.

Okay, tested out this:

Performance IS the same compared to 5.5.5 on the old Cryptonight algorithm. 
One thing that i noticed is, that i instantly got 2 Rejected shares for some reason. Also, way slower start and speed ramping. But it works so far without issues.

The GPU monitoring is still broken - as always. GPU positions are mixed (GPU2 monitoring is on position 0, but speed stats are on position 2).

**Comment by ku4eto** [comment]
Update:

I didn't realize, that i need to explicitly use the "monero":true and a .conf file in order to work with the new PoW. If you don't use it, it will split 1-2 Rejected shares and the rest will be Accepted, BUT they are not accepted in reality by the pool.

So you create an .conf file, and in the POOLS header you specify this argument.

Example below:

{
    "pools": [{
        "url": "POOL:PORT",
        "user": "WALLET",
        "pass": "x",
        "monero":true,
        "profile" : "xmr"
    }
],
    "profiles": [{
            "worksize": "8",
            "name": "xmr",

            "algorithm": "cryptonight",
            "gpu-threads": "2",
            "rawintensity": "896",
            "device": "2"
        }
    ],
    "no-extranonce": "false",
    "default-profile": "xmr"
}

Performance is still about the same - 1073h/s. Maybe about 5h/s loss of performance, but its always questionable.

**Comment by dannycoin** [comment]
this does not work with another coin with monero7 like Stellite XTL.. only work with MONERO (XMR).

**Comment by LightningW9** [comment]
Yes it was designed to auto switch Monero, so it always reads the version from the block header before switching if it is V7. It looks like Stellite block version is V3 although they are using the updated Monero V7 algorithm.

There is a pull request created by ystarnaud to add support for more cryptonight coins.

Probably the block header version checking could be removed as it a bit redundant now Monero switched. The manual switch with "monero":true would be enough to change to the new variant.

**Comment by ku4eto** [comment]
Graft just did a fork to V7. Tested and it works normally.
Using a config file with             "monero":true, in it.
On start of the miner, it says

Switching to variant 1.
Shares are accepted then.

**Comment by pass75** [comment]
Well to know. Like LightningW9 said it seems it's just a matter of block header. Maybe Graft is using v7 header like Monero so the sgminer switches to variant 1 correctly. Thanks to notice us.

**Comment by ku4eto** [comment]
From what i know, Graft PoW change is written as v8 as seen from here:
https://github.com/graft-project/GraftNetwork#graft-network-software-updates-and-consensus-protocol-changes-hard-fork-schedule

If it works, it works.

**Comment by ku4eto** [comment]
The .bat file should point to a .conf file. Create a .txt file and change the extension to .conf.
Example:

XMR.bat
--------
sgminer.exe -c All_cards.conf
--------

All_cards.conf
--------

{
    "pools": [{
        "url": "POOL_PORT",
        "user": "WALLET",
        "pass": "x",
        "monero": true
        "profile" : "xmr"
    }
],
    "profiles": [{
            "worksize": "8,4",
            "name": "xmr",
            "algorithm": "cryptonight",
            "gpu-threads": "2",
            "rawintensity": "896,256",
            "device": "0,1"
        }
    ],
    "default-profile": "xmr"
}

**Comment by ku4eto** [comment]
@LightningW9  Heads up everyone.
The 5.5.6 build actuallly WORKS with Ethash. The issue is, if you try to create the binaries upon first time starting. Simply copy over the ones from the 5.5.5.
I mean those files: ethashEllesmeregw128l4.bin

**Comment by jayanh** [comment]
@LightningW9 Could you tell me the steps to build on window using mingw. I follow the guide but it show error about conflict "sleep" in compact.h.

**Comment by afrodisio99** [comment]
@LightningW9  can support linux ...

**Comment by jayanh** [comment]
> @LightningW9 can support linux ...

Hi! could you help me on linux. Could i contact you from telegram, this is my: https://t.me/inlasy
Thanks
